homeassistant/components/hue/__init__.py

```diff
@@ -103,49 +85,73 @@
+                continue
+
+            # Run through config schema to populate defaults
+            bridges.append(BRIDGE_CONFIG_SCHEMA({
+                CONF_HOST: entry['internalipaddress'],
+                # Careful with using entry['id'] for other reasons. The
+                # value is in lowercase but is returned uppercase from hub.
+                CONF_FILENAME: '.hue_{}.conf'.format(entry['id']),
+            }))
     else:
         # Component not specified in config, we're loaded via discovery
         bridges = []
 
     if not bridges:
         return True
 
-    await asyncio.wait([
-        async_setup_bridge(
-            hass, bridge[CONF_HOST], bridge[CONF_FILENAME],
-            bridge[CONF_ALLOW_UNREACHABLE], bridge[CONF_ALLOW_HUE_GROUPS]
-        ) for bridge in bridges
-    ])
+    for bridge_conf in bridges:
+        host = bridge_conf[CONF_HOST]
 
-    return True
+        # Store config in hass.data so the config entry can find it
+        hass.data[DOMAIN][host] = bridge_conf
+
+        # If configured, the bridge will be set up during config entry phase
+        if host in configured:
+            continue
+
+        # No existing config entry found, try importing it or trigger link
+        # config flow if no existing auth. Because we're inside the setup of
+        # this component we'll have to use hass.async_add_job to avoid a
+        # deadlock: creating a config entry will set up the component but the
+        # setup would block till the entry is created!
+        hass.async_add_job(hass.config_entries.flow.async_init(
+            DOMAIN, source='import', data={
+                'host': bridge_conf[CONF_HOST],
+                'path': bridge_conf[CONF_FILENAME],
+            }
+        ))
 
+    return True
 
-async def async_setup_bridge(
-        hass, host, filename=None,
-        allow_unreachable=DEFAULT_ALLOW_UNREACHABLE,
-        allow_hue_groups=DEFAULT_ALLOW_HUE_GROUPS,
-        username=None):
-    """Set up a given Hue bridge."""
-    assert filename or username, 'Need to pass at least a username or filename'
 
-    # Only register a device once
-    if host in hass.data[DOMAIN]:
-        return
+async def async_setup_entry(hass, entry):
+    """Set up a bridge from a config entry."""
+    host = entry.data['host']
+    config = hass.data[DOMAIN].get(host)
 
-    if username is None:
-        username = await hass.async_add_job(
-            _find_username_from_config, hass, filename)
+    if config is None:
+        allow_unreachable = DEFAULT_ALLOW_UNREACHABLE
+        allow_groups = DEFAULT_ALLOW_HUE_GROUPS
+    else:
+        allow_unreachable = config[CONF_ALLOW_UNREACHABLE]
+        allow_groups = config[CONF_ALLOW_HUE_GROUPS]
 
-    bridge = HueBridge(host, hass, filename, username, allow_unreachable,
-                       allow_hue_groups)
-    await bridge.async_setup()
+    bridge = HueBridge(hass, entry, allow_unreachable, allow_groups)
     hass.data[DOMAIN][host] = bridge
```

Comment:
Is it intended to overwrite the bridge config in `hass.data[host] ` with the bridge instance?

Reply:
Yeah. The setup will store the config for the entries and then bridge instance is set via config entry. If there is no auth, setup will initiate config flow to configure entry and once finished, the created config entry will pick up the config